### PR TITLE
Introduce foreman_tasks_core (RPM)

### PR DESCRIPTION
**Pull request by iNecas**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [x] 1.13
* [ ] 1.12
* [ ] 1.11

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

Since https://github.com/theforeman/foreman-packaging/pull/1321
contains multiple related packages, that one depends on others,
I'm splitting it to PRs to be handled on by one. This packages
are the first to get in.

### CODE REVIEW

File: rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.1.0
+Release: 1%{?foremandist}%{?dist}
+Group: Development/Languages
+License: GPLv3
+URL: http://github.com/theforeman/foreman-tasks
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+%if 0%{?el6} && 0%{!?scl:1}
+Requires: %{?scl_prefix_ruby}ruby(abi)
```
Comment (domcleal) [inline rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec]: Unused block, remove it

File: rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+%global gem_name foreman-tasks-core
+
+Summary: Code used both at Forman and Foreman proxy regarding tasks
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.1.0
+Release: 1%{?foremandist}%{?dist}
+Group: Development/Languages
+License: GPLv3
+URL: http://github.com/theforeman/foreman-tasks
```
Comment (domcleal) [inline rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec]: https, per gemspec

File: rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name foreman-tasks-core
+
+Summary: Code used both at Forman and Foreman proxy regarding tasks
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.1.0
+Release: 1%{?foremandist}%{?dist}
+Group: Development/Languages
+License: GPLv3
```
Comment (domcleal) [inline rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec]: Missing a LICENSE file in the gem.

File: rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+
+%description
+Common code used both at Forman and Foreman proxy regarding tasks
+
+%package doc
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
Comment (domcleal) [inline rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec]: No such package, remove this line.

**Comment by iNecas** [comment]
Updated based on suggestions

File: rubygem-dynflow/rubygem-dynflow.spec [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
 Summary: DYNamic workFLOW engine
 Name: %{?scl_prefix}rubygem-%{gem_name}
-Version: 0.8.13
+Version: 0.8.15
 Release: 1%{?foremandist}%{?dist}
 Group: Development/Languages
 License: MIT
 URL: https://github.com/Dynflow/dynflow
-Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
```
Comment (domcleal) [inline rubygem-dynflow/rubygem-dynflow.spec]: Use an HTTP source here for now, git-annex on our builders don't currently support HTTPS (should be fixed soon once rebuilt). The previous comment was only about the URL field, not Source0.

File: rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+%global gem_name foreman-tasks-core
+
+Summary: Code used both at Forman and Foreman proxy regarding tasks
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.1.1
+Release: 1%{?foremandist}%{?dist}
+Group: Development/Languages
+License: GPLv3
+URL: https://github.com/theforeman/foreman-tasks
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
```
Comment (domcleal) [inline rubygem-foreman-tasks-core/rubygem-foreman-tasks-core.spec]: ditto, Source0 should be HTTP for now. The SHA256 sum is stored by git-annex in the repo so the downloaded file is still verified.

File: rubygem-dynflow/rubygem-dynflow.spec (hunk 5 above)
Comment (iNecas) [inline rubygem-dynflow/rubygem-dynflow.spec]: Oh, I though ti was for both, reverted this changes then

**Comment by iNecas** [comment]
I've tested this RPMs and they seem to work ok so far

**Comment by domcleal** [comment]
Merged and built on both rpm/develop and rpm/1.13, thanks @iNecas.

**Pull request merged**